lib/scholar/metrics/similarity.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Scholar.Metrics.Similarity do
+  @moduledoc """
+  Similarity metrics between 1-D tensors.
+  """
+
+  import Nx.Defn
+  import Scholar.Shared
+
+  @doc """
+  The Jaccard similarity (also known as Jaccard similarity coefficient, or Jaccard index)
+  is a statistic used to measure similarities between two sets. Mathematically, the calculation
+  of Jaccard similarity is the ratio of set intersection over set union.
+
+  J(A, B) = |A∩B| / |A∪B|
+
+  ## Examples
+
+      iex> x = Nx.tensor([1.0, 5.0, 3.0, 6.7])
+      iex> y = Nx.tensor([5.0, 2.5, 3.1, 9.0])
+      iex> Scholar.Metrics.Similarity.jaccard(x, y)
+      #Nx.Tensor<
+        f32
+        0.1428571492433548
+      >
+
+      iex> x = Nx.tensor([1,2,3,5,7])
+      iex> y = Nx.tensor([1,2,4,8,9])
+      iex> Scholar.Metrics.Similarity.jaccard x, y
+      #Nx.Tensor<
+        f32
+        0.25
+      >
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([1, 2, 3])
+      iex> Scholar.Metrics.Similarity.jaccard x, y
+      ** (ArgumentError) expected input shapes to be equal, got {2} != {3}
+  """
+  defn jaccard(x, y) do
+    assert_same_shape!(x, y)
```

Comment:
Perhaps it would be a good idea to support rank N tensors where the similarity is calculated over the last axis?

Reply:
I can take a look during the week.

Reply:
With this proposal, the shape would need to be the same except on the last axis, I think

Reply:
Not sure how to do it here. What's your suggestion @polvalente ?

Reply:
It appears same shape is required here: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.jaccard_score.html

but it allows multi dimensions.

Reply:
We can keep requiring the same shape. Please add a comment similar to: "We're requiring the same shape because usual use cases will have the same shape. The last axis could in theory be different on both sides"

Reply:
comment added